[Lin Chenyu] iP

LINCHENYU2030S opened the pull request:
# Alex
> *“Your mind is for having ideas, not holding them.” – David Allen* [source](https://dansilvestre.com/productivity-quotes)

## Alex frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* FAST SUPER FAST to use

## The commands include
1. add task
2. delete task
3. find / view task
4. edit task

## Here is the `main` method:
```java
public static void main(String[] args) {
        new Alex().run();
}

## Features
- [x] manage todo task
- [x] manage deadline task
- [x] manage event task
- [ ] GUI (coming soon) 😸

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

LINCHENYU2030S committed:
Rename, Greet, Exit

LINCHENYU2030S committed:
Echo

LINCHENYU2030S committed:
Add, List

LINCHENYU2030S committed:
Mark, Unmark

LINCHENYU2030S committed:
ToDos, Events, Deadlines

LINCHENYU2030S committed:
Automated Text UI Testing

LINCHENYU2030S committed:
Delete

LINCHENYU2030S committed:
Handle Errors

LINCHENYU2030S committed:
add saving user data feature

LINCHENYU2030S committed:
add saving feature

LINCHENYU2030S committed:
add saving user data features

LINCHENYU2030S committed:
Merge branch 'branch-Level-7'

LINCHENYU2030S committed:
print Dates and Times, print Task on given Date

LINCHENYU2030S committed:
Merge branch 'branch-Level-8'

LINCHENYU2030S committed:
More OOP

LINCHENYU2030S committed:
apply more OOP

LINCHENYU2030S committed:
optimize code

LINCHENYU2030S committed:
Organize classes into packages

LINCHENYU2030S committed:
Merge branch 'branch-A-Packages'

LINCHENYU2030S committed:
Merge branch 'add-gradle-support'

LINCHENYU2030S committed:
set up gradle

LINCHENYU2030S committed:
add JUnit Tests

LINCHENYU2030S committed:
create a JAR File

LINCHENYU2030S committed:
optimize code

LINCHENYU2030S committed:
add find matching task(s) feature

LINCHENYU2030S committed:
Merge branch 'branch-Level-9'

zannloo commented on src/main/java/Alex/AddCommand.java:
```diff
@@ -0,0 +1,109 @@
+
+public class AddCommand extends Command{
+    private String command;
+    private Add addType;
```
I like how the variable names are in camelCase.

zannloo commented on src/main/java/Alex/Alex.java:
```diff
@@ -0,0 +1,17 @@
+package Alex;
```
I like how every class is in a package.

zannloo commented on src/main/java/Alex/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+        return this.by;
+    }
+
+    public LocalDate getDueDate() {
```
I like how names representing methods are verbs and written in camelCase.

zannloo commented on src/main/java/Alex/AddCommand.java:
```diff
@@ -0,0 +1,109 @@
+    @Override
+    public void execute() {
+        switch(this.addType) {
+            case TODO:
```
Not sure if I like that there is an indentation for case clauses.

zannloo commented on src/main/java/Alex/ExitCommand.java:
```diff
@@ -0,0 +1,14 @@
+package Alex;
+
+public class ExitCommand extends Command{
```
Not sure if I like that there was no space before '{'.

zannloo commented on src/main/java/Alex/UserInputStorage.java:
```diff
@@ -0,0 +1,199 @@
+
+public class UserInputStorage {
+
+    public static final DateTimeFormatter TIMEFORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
```
I like how the constant name is all uppercased but I am not sure if timeformatter is one word or two.

zannloo reviewed (commented):
LGTM:) Overall, I like how well organised your project was and that the naming of variables and methods were really consistent.

LINCHENYU2030S committed:
testing

LINCHENYU2030S committed:
comply with coding standard

LINCHENYU2030S committed:
resolve conflict fromm branch-A-CodingStandard

LINCHENYU2030S committed:
add more changes to comply with coding standard

LINCHENYU2030S committed:
add JavaDoc

LINCHENYU2030S committed:
resolve conflict when merging branch-A-JavaDoc

LINCHENYU2030S committed:
add checkstyle config

LINCHENYU2030S committed:
Merge branch 'branch-A-CheckStyle'

LINCHENYU2030S committed:
some minor changes

LINCHENYU2030S committed:
add GUI

LINCHENYU2030S committed:
Merge branch 'branch-Level-10'

LINCHENYU2030S committed:
package more

LINCHENYU2030S committed:
modify build.gradle
- add shadow to build.gradle for fat jar file generation
- rename the jar file generated in build.gradle

LINCHENYU2030S committed:
add more packages
- add a package called commands
- add a package called storage
- add a package called tasks
- add a package called ui

LINCHENYU2030S committed:
add assertions

LINCHENYU2030S committed:
improve code quality

LINCHENYU2030S committed:
Merge pull request #2 from LINCHENYU2030S/branch-A-Assertions

add assertions

LINCHENYU2030S committed:
Merge branch 'master' into branch-A-CodeQuality

LINCHENYU2030S committed:
avoid long method

LINCHENYU2030S committed:
Merge pull request #3 from LINCHENYU2030S/branch-A-CodeQuality

improve code quality

LINCHENYU2030S committed:
set up CI

LINCHENYU2030S committed:
add help extension

LINCHENYU2030S committed:
Merge branch 'branch-BCD-Extension'

LINCHENYU2030S committed:
added more exception handling

LINCHENYU2030S committed:
Merge branch 'branch-A-MoreErrorHandling'

LINCHENYU2030S committed:
update user guide and Ui.png

LINCHENYU2030S committed:
remove .github/workflow temporarily

LINCHENYU2030S committed:
remove .github/workflow temporarily

LINCHENYU2030S committed:
edit build.gradle

LINCHENYU2030S committed:
add back CI
